packages/edit-site/src/components/page-actions/trash-page-menu-item.js

```diff
@@ -1,36 +1,35 @@
 /**
  * WordPress dependencies
  */
-import { useDispatch, useSelect } from '@wordpress/data';
+import { useDispatch } from '@wordpress/data';
 import { decodeEntities } from '@wordpress/html-entities';
 import { store as coreStore } from '@wordpress/core-data';
 import { __, sprintf } from '@wordpress/i18n';
 import { MenuItem } from '@wordpress/components';
 import { store as noticesStore } from '@wordpress/notices';
 
-export default function TrashPageMenuItem( { postId, onRemove } ) {
+export default function TrashPageMenuItem( { page, onRemove } ) {
 	const { createSuccessNotice, createErrorNotice } =
 		useDispatch( noticesStore );
 	const { deleteEntityRecord } = useDispatch( coreStore );
-	const page = useSelect(
-		( select ) =>
-			select( coreStore ).getEntityRecord( 'postType', 'page', postId ),
-		[ postId ]
+	const title = decodeEntities(
+		typeof page.title === 'string' ? page.title : page.title.rendered
```

Comment:
When is `page.title` not an object? I wish `@wordpress/data` handled this for us.

Reply:
> When is page.title not an object?

When consuming an edited record, [raw properties](https://github.com/WordPress/gutenberg/blob/4cfaab6302c862ce264247ebef31b45729f04343/packages/core-data/src/entities.js#L21) will be strings. It also depends on the REST API schema, but it should be the same for post types.

> I wish @wordpress/data handled this for us.

It sort of does, but I always have a problem remembering which selector returns which data shape 😓

Reply:
Good question 👍 

As George notes, this should now always be getting the edited record so the `title` value should be a string. I'll update this PR accordingly.

Reply:
I was mistaken, it appears that the sidebar nav screen for pages is [not using the edited record](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/sidebar-navigation-screen-page/index.js?plain=1#L41).

For the moment, this trash menu item is getting a record with a string value when accessed via the page panel but an object value in the site editor's sidebar nav. I'd like to explore consistently using the edited record in a follow-up.